pkg/model/application_live_state.go

```diff
@@ -60,23 +60,46 @@ func (s KubernetesResourceState) HasDiff(a KubernetesResourceState) bool {
 }
 
 // DetermineAppHealthStatus updates its own health status, which is determined based on its resources status.
+// TODO: Determine health state of other than k8s and cloud run app
 func (s *ApplicationLiveStateSnapshot) DetermineAppHealthStatus() {
 	switch s.Kind {
+	case ApplicationKind_TERRAFORM,
+		ApplicationKind_LAMBDA,
+		ApplicationKind_ECS:
 	case ApplicationKind_KUBERNETES:
-		k := s.Kubernetes
-		if k == nil {
-			return
-		}
-		status := ApplicationLiveStateSnapshot_HEALTHY
-		for _, r := range k.Resources {
-			if r.HealthStatus == KubernetesResourceState_OTHER {
-				status = ApplicationLiveStateSnapshot_OTHER
-				break
-			}
+		s.determineKubernetesAppHealthStatus()
+	case ApplicationKind_CLOUDRUN:
+		s.determineCloudRunAppHealthStatus()
+	}
+	return
+}
+
+func (s *ApplicationLiveStateSnapshot) determineKubernetesAppHealthStatus() {
+	k := s.Kubernetes
+	if k == nil {
+		return
+	}
+	status := ApplicationLiveStateSnapshot_HEALTHY
+	for _, r := range k.Resources {
+		if r.HealthStatus == KubernetesResourceState_OTHER {
+			status = ApplicationLiveStateSnapshot_OTHER
+			break
 		}
-		s.HealthStatus = status
-	default:
-		// TODO: Determine health state of other than k8s app
+	}
+	s.HealthStatus = status
+}
+
+func (s *ApplicationLiveStateSnapshot) determineCloudRunAppHealthStatus() {
+	c := s.Cloudrun
```

Comment:
nit: IMO, "app" is more understandable than "c" 🤚

Reply:
TYVM:D
PTAL https://github.com/pipe-cd/pipecd/pull/3280/commits/ee9ff7d8edfc9ad366d45ce29c83d950a03625a5 when you can check it.